crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -523,3 +555,36 @@ pub(crate) fn unassigned_special_variable_in_stub(
         stmt.range(),
     ));
 }
+
+/// PIY026
+pub(crate) fn type_alias_check(checker: &mut Checker, value: &Expr, targets: &[Expr]) {
+    let is_any = |expr: &Expr| {
+        let Expr::Name(ast::ExprName { id, .. }) = expr else {
+            return false;
+        };
+        id == "Any"
+    };
+
+    let target = &targets[0];
+    if is_special_assignment(target, checker.semantic()) {
+        return;
+    }
+    if matches!(value, Expr::Name(_)) && !is_any(value) {
+        return;
+    }
```

Comment:
I think more idiomatic in our codebase would be to remove `is_any`, and do:

```rust
if let Expr::Name(ast::ExprName { id, .. }) = target {
    if id.as_str() == "Any" {
        return;
    }
};
```

That way, IMO, the condition reads more linearly: "if the `value` is a `Name`, and that `Name` is "any", then return".

Reply:
However, we have a special utility for this, which will ensure that the `Any` here actually refers to `typing.Any`, and that we match things like `x: typing.Any` instead of just `x: Any`:

```rust
if checker.semantic().match_typing_expr(value, "Any") {
    return;
}
```